### ISSUE TITLE
[Bug?]: rw setup auth auth0 installing latest version of auth0-spa-js

### ISSUE THREAD

**Issue by tbay06** [redwoodjs/redwood#6961]
### What's not working?

`auth0/auth0-spa-js` bumped their SemVer to [v2.0.0](https://github.com/auth0/auth0-spa-js/releases/tag/v2.0.0) last month with some [breaking changes](https://github.com/auth0/auth0-spa-js/blob/master/MIGRATION_GUIDE.md). The package installations for the auth setup aren't version locked when the command is executed. Possibly related to pull request [#6956](https://github.com/redwoodjs/redwood/pull/6956) and pull request [#6922](https://github.com/redwoodjs/redwood/pull/6922)

The main breaking change I noticed requires an update to the `auth-providers-setup/src/auth0/templates/web/auth.ts.template` to conform to the new changes in the instantiation of the `Auth0Client`, to go from this:

```typescript
import { Auth0Client } from '@auth0/auth0-spa-js'

import { createAuth0Auth } from '@redwoodjs/auth-providers-web'

const auth0 = new Auth0Client({
  domain: process.env.AUTH0_DOMAIN || '',
  client_id: process.env.AUTH0_CLIENT_ID || '',
  redirect_uri: process.env.AUTH0_REDIRECT_URI,

  // ** NOTE ** Storing tokens in browser local storage provides persistence across page refreshes and browser tabs.
  // However, if an attacker can achieve running JavaScript in the SPA using a cross-site scripting (XSS) attack,
  // they can retrieve the tokens stored in local storage.
  // https://auth0.com/docs/libraries/auth0-spa-js#change-storage-options
  cacheLocation: 'localstorage',
  audience: process.env.AUTH0_AUDIENCE,

  // @MARK useRefreshTokens is required for automatically extending sessions
  // beyond that set in the initial JWT expiration.
  //
  // @see https://auth0.com/docs/tokens/refresh-tokens
  // useRefreshTokens: true,
})

export const { AuthProvider, useAuth } = createAuth0Auth(auth0)
```

to this:

```typescript
import { Auth0Client } from '@auth0/auth0-spa-js'

import { createAuth0Auth } from '@redwoodjs/auth-providers-web'

const auth0 = new Auth0Client({
  domain: process.env.AUTH0_DOMAIN || '',
  clientId: process.env.AUTH0_CLIENT_ID || '',
  authorizationParams:{
     redirect_uri: process.env.AUTH0_REDIRECT_URI,
     audience: process.env.AUTH0_AUDIENCE
  }
  // ** NOTE ** Storing tokens in browser local storage provides persistence across page refreshes and browser tabs.
  // However, if an attacker can achieve running JavaScript in the SPA using a cross-site scripting (XSS) attack,
  // they can retrieve the tokens stored in local storage.
  // https://auth0.com/docs/libraries/auth0-spa-js#change-storage-options
  cacheLocation: 'localstorage',
  

  // @MARK useRefreshTokens is required for automatically extending sessions
  // beyond that set in the initial JWT expiration.
  //
  // @see https://auth0.com/docs/tokens/refresh-tokens
  // useRefreshTokens: true,
})

export const { AuthProvider, useAuth } = createAuth0Auth(auth0)
```
where `client_id` gets changed to `clientId` and the addition of the `authorizationParams` object.

This fixes the breaking change with `auth0` auth in new projects, but there are changes to the `logout` method in v2.0.0 that may need to be addressed as well.

### How do we reproduce the bug?

Run `yarn rw auth setup auth0` in a fresh redwood project see the errors in `web/src/App.tsx`. This also does not allow the user to log into auth0, with a `something went wrong...` error displaying from auth0 on login attempts.

### What's your environment? (If it applies)

```shell
System:
    OS: Windows 10 10.0.19044
  Binaries:
    Node: 16.14.2 - ~\AppData\Local\Temp\xfs-65792ac0\node.CMD
    Yarn: 3.2.3 - ~\AppData\Local\Temp\xfs-65792ac0\yarn.CMD
  Browsers:
    Chrome: 107.0.5304.107
    Edge: Spartan (44.19041.1266.0), Chromium (107.0.1418.56)
  npmPackages:
    @redwoodjs/core: 3.5.0 => 3.5.0
```

### Are you interested in working on this?

- [X] I'm interested in working on this

**Comment by Tobbe** [comment]
Thanks for the bug report @tbay06
I pinned the setup script to v1 for now. But we should definitely upgrade to v2 at some point, so I'm leaving this issue open so we don't forget. 
In the meantime it'd be great if you could figure out what changes (if any) we need to do related to `logout`.

**Comment by tbay06** [comment]
Awesome, thanks for the quick fix @Tobbe! I'll dig into it a little more and see where else we need to make code mods to work with v2.

**Comment by Tobbe** [comment]
I implemented the fix in #6922 but to get it out faster I now also implemented it on our "next" branch https://github.com/redwoodjs/redwood/pull/6963

**Comment by jtoar** [comment]
This one was fixed in https://github.com/redwoodjs/redwood/pull/6963 and went out in [v3.6.1](https://github.com/redwoodjs/redwood/releases/tag/v3.6.1)
